shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/Spark311Shims.scala

```diff
@@ -275,7 +275,7 @@ class Spark311Shims extends Spark301Shims {
         }),
       GpuOverrides.exec[InMemoryTableScanExec](
         "Implementation of InMemoryTableScanExec to use GPU accelerated Caching",
-        ExecChecks(TypeSig.commonCudfTypes + TypeSig.DECIMAL, TypeSig.all),
+        ExecChecks(TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.STRUCT, TypeSig.all),
```

Comment:
Shouldn't this be something like:
```suggestion
        ExecChecks((TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.STRUCT).nested(), TypeSig.all),
```

Reply:
It is not very clear to me. Do we have documentation on this?

Reply:
The Scaladoc for TypeChecks, etc. is the documentation.  It should be clear whether you need it or not based on the generated docs which are missing in this PR.  I suspect without using `nested()` the docs will say STRUCT is supported but not any of the types within that struct.

Reply:
I used to be able to gen docs by invoking `mvn build` thats not happening now.

Reply:
`build` is not part of the Maven default lifecycle goals.  You should be running `mvn verify` to perform the doc generation per [the pom file](https://github.com/NVIDIA/spark-rapids/blob/branch-0.5/dist/pom.xml#L162-L180) and [contributing guide](https://github.com/NVIDIA/spark-rapids/blob/branch-0.5/CONTRIBUTING.md#building-from-source).

Reply:
sorry, I meant to write `mvn install`

I think something broke in my workflow because I used to see the docs generated by simply running my script. Will update and get back

Reply:
The reason why the docs are not generated is because they aren't auto-generated for the Shims. If I manually generate it, where do the docs go for the Shims?
That could be a bigger effort, should we may be consider handling it as a separate PR?

Reply:
Yes, we should handle doc generation for stuff that requires a minimum Spark version beyond the base supported Spark version as a separate issue.  I filed #1977 to track.